equinix/resource_metal_organization_member.go

```diff
@@ -0,0 +1,277 @@
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceMetalOrganizationMemberCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).metal
+
+	email := d.Get("invitee").(string)
+	createRequest := &packngo.InvitationCreateRequest{
+		Invitee:     email,
+		Roles:       convertStringArr(d.Get("roles").(*schema.Set).List()),
+		ProjectsIDs: convertStringArr(d.Get("projects_ids").(*schema.Set).List()),
+		Message:     strings.TrimSpace(d.Get("message").(string)),
+	}
+
+	orgID := d.Get("organization_id").(string)
+	_, _, err := client.Invitations.Create(orgID, createRequest, nil)
+	if err != nil {
+		return friendlyError(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", email, orgID))
+
+	return resourceMetalOrganizationMemberRead(d, meta)
+}
+
+func findMember(invitee string, members []packngo.Member, invitations []packngo.Invitation) (*member, error) {
+	for _, mbr := range members {
+		if mbr.User.Email == invitee {
+			return &member{Member: &mbr}, nil
+		}
+	}
+
+	for _, inv := range invitations {
+		if inv.Invitee == invitee {
+			return &member{Invitation: &inv}, nil
+		}
+	}
+	return nil, fmt.Errorf("member not found")
+}
+
+func resourceMetalOrganizationMemberRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).metal
+	parts := strings.Split(d.Id(), ":")
+	invitee := parts[0]
+	orgID := parts[1]
+
+	// orgID := d.Get("organization_id").(string)
+	// invitee := d.Get("invitee").(string)
+
+	listOpts := &packngo.ListOptions{Includes: []string{"user"}}
+	invitations, _, err := client.Invitations.List(orgID, listOpts)
+	if err != nil {
+		err = friendlyError(err)
+		// If the org was destroyed, mark as gone.
+		if isNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	org, _, err := client.Organizations.Get(orgID, &packngo.GetOptions{Includes: []string{"members", "members.user"}})
+	if err != nil {
+		err = friendlyError(err)
+		// If the org was destroyed, mark as gone.
+		if isNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	member, err := findMember(invitee, org.Members, invitations)
+	if !d.IsNewResource() && err != nil {
+		return friendlyError(fmt.Errorf("Could not find member %s in organization %s", invitee, d.Get("organization_id").(string)))
+	}
+
+	if member.isMember() {
+		projectIDs := []string{}
+		for _, project := range member.Member.Projects {
+			projectIDs = append(projectIDs, path.Base(project.URL))
+		}
+		return setMap(d, map[string]interface{}{
+			"state":           "active",
+			"roles":           stringArrToIfArr(member.Member.Roles),
+			"projects_ids":    stringArrToIfArr(projectIDs),
+			"organization_id": member.Member.Organization.ID,
+		})
+	} else if member.isInvitation() {
+		projectIDs := []string{}
+		for _, project := range member.Invitation.Projects {
+			projectIDs = append(projectIDs, path.Base(project.Href))
+		}
+		return setMap(d, map[string]interface{}{
```

Comment:
If `invited_by` is computed I expect it here in the setMap

Reply:
invited_by is a user in the database. We may want to change this field to a user_id value.

Reply:
updated to user_id in 9eee41d